xdsl/utils/lexer.py

```diff
@@ -31,31 +31,34 @@ def __post_init__(self):
     def __len__(self):
         return self.len
 
-    def get_lines_containing(self, span: Span) -> tuple[list[str], int, int] | None:
-        # A pointer to the start of the first line
-        start = 0
-        line_no = span.line_offset
+    def get_start_of_line(self, pos: Position) -> Position:
+        """
+        Returns the location of the last newline before `pos`, or 0 if there are no
+        previous newlines.
+        """
+        # Returns -1 if not found, meaning on first line
+        pos = self.content.rfind("\n", 0, pos)
+        # If the result is not the beginning of the input, go past the matching newline
+        return pos + 1
+
+    def get_end_of_line(self, pos: Position) -> Position:
+        """
+        Returns the position of the first newline after `pos`, or the length of the
+        file if there are no more newlines.
+        """
+        pos = self.content.find("\n", pos)
+        # If the result is at the end of the input, return the length for correct slice
+        # indexing
+        return self.len if pos == -1 else pos
+
+    def get_lines_containing(self, span: Span) -> tuple[list[str], int, int]:
```

Comment:
This needs a docstring in my opinion, what does the second int mean?

Reply:
I would like to remove it in 1/2 PRs